File: 2427-number-of-common-factors/2427-number-of-common-factors.py
from math import *
def cnt(n):
    cnt = 0
    for i in range(1, isqrt(n) + 1):
        if n % i == 0:
            cnt += 1
            if i != n // i: cnt += 1
    return cnt
class Solution:
    def commonFactors(self, a: int, b: int) -> int:
        g = gcd(a, b)
        return cnt(g)

File: 2427-number-of-common-factors/2427-number-of-common-factors.cpp
class Solution {
public:
    int cnt(int n){
        int c = 0;
        for (int i = 1; i <= sqrt(n); ++i){
            if (n % i == 0){
                ++c;
                if (i != n / i) ++c;
            }
        }
        return c;
    }
    int commonFactors(int a, int b) {
        int g = __gcd(a, b);
        return cnt(g);
    }
};